openSearchLane.go

```diff
@@ -0,0 +1,469 @@
+			offline: true,
+		}
+	}
+
+	err := osl.connect(config)
+	if err != nil {
+		osl.Error(err)
+	}
+
+	go osl.openSearchConnection.processConnection()
+
+	l = &osl
+
+	return
+
+}
+
+func (osl *openSearchLane) connect(config *OslConfig) (err error) {
+	var client *opensearchapi.Client
+
+	if !config.offline && osl.openSearchConnection.client != nil && osl.openSearchConnection.client.Client != nil {
+		osl.openSearchConnection.wakeCh <- true
+	}
+
+	if !config.offline {
+		client, err = newOpenSearchClient(config.OpenSearchUrl, config.OpenSearchPort, config.OpenSearchUser, config.OpenSearchPass, config.OpenSearchTransport)
+		if err != nil {
+			return
+		}
+	}
+	osl.openSearchConnection.client = client
+	osl.openSearchConnection.config = config
+
+	if config.LogThreshold == 0 {
+		osl.openSearchConnection.config.LogThreshold = OslDefaultLogThreshold
+	}
+	if config.MaxBufferSize == 0 {
+		osl.openSearchConnection.config.MaxBufferSize = OslDefaultMaxBufferSize
+	}
+	if config.BackoffInterval == 0 {
+		osl.openSearchConnection.config.BackoffInterval = OslDefaultBackoffInterval
+	}
+	if config.BackOffLimit == 0 {
+		osl.openSearchConnection.config.BackOffLimit = OslDefaultBackOffLimit
+	}
+
+	return
+
+}
+
+func (osl *openSearchLane) ReConnect(config *OslConfig) (err error) {
+	return osl.connect(config)
+}
+
+func (osl *openSearchLane) Derive() Lane {
+	ll := deriveLogLane(&osl.logLane, context.WithValue(osl.Context, parent_lane_id, osl.LaneId()), osl.tees, osl.cr)
+
+	osl2 := finishDerive(osl, ll)
+
+	return osl2
+}
+
+func (osl *openSearchLane) DeriveWithCancel() (Lane, context.CancelFunc) {
+	childCtx, cancelFn := context.WithCancel(context.WithValue(osl.logLane.Context, parent_lane_id, osl.logLane.LaneId()))
+	ll := deriveLogLane(&osl.logLane, childCtx, osl.tees, osl.cr)
+
+	osl2 := finishDerive(osl, ll)
+
+	return osl2, cancelFn
+}
+
+func (osl *openSearchLane) DeriveReplaceContext(ctx context.Context) Lane {
+	ll := deriveLogLane(&osl.logLane, context.WithValue(ctx, parent_lane_id, osl.LaneId()), osl.tees, osl.cr)
+
+	osl2 := finishDerive(osl, ll)
+
+	return osl2
+}
+
+func finishDerive(osl *openSearchLane, ll *logLane) *openSearchLane {
+	osl2 := openSearchLane{
+		openSearchConnection: osl.openSearchConnection,
+	}
+	osl.openSearchConnection.attach()
+
+	ll.setFlagsMask(log.Ldate | log.Ltime)
+	ll.clone(&osl2.logLane)
+	osl2.logLane.writer = log.New(&osl2, "", 0)
+
+	return &osl2
+}
+
+func (osl *openSearchLane) SetMetadata(key string, val string) {
+	osl.openSearchConnection.mu.Lock()
+	defer osl.openSearchConnection.mu.Unlock()
+
+	if osl.metadata == nil {
+		osl.metadata = make(map[string]string)
+	}
+
+	osl.metadata[key] = val
+}
+
+func (osl *openSearchLane) Close() {
+	osl.openSearchConnection.detach()
+}
+
+func (osl *openSearchLane) Write(p []byte) (n int, err error) {
+	osl.openSearchConnection.mu.Lock()
+	defer osl.openSearchConnection.mu.Unlock()
+
+	logEntry := string(p)
+
+	if strings.Contains(logEntry, "\n") {
```

Comment:
No need to test contains, just do the ReplaceAll. Probably should replace it with a space, not empty string